parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -526,7 +528,35 @@ public PageReadStore readNextRowGroup() throws IOException {
     return columnChunkPageReadStore;
   }
 
+  public static DictionaryPage readDictionary(Configuration conf, ColumnChunkMetaData meta, FSDataInputStream fin) throws IOException {
+    if(fin.getPos() != meta.getStartingPos()) {
+      fin.seek(meta.getStartingPos());
+    }
+
+    PageHeader pageHeader = Util.readPageHeader(fin);
+
+    return readDictionary(conf, pageHeader, meta, fin);
+  }
+
+  public static DictionaryPage readDictionary(Configuration conf, PageHeader pageHeader, ColumnChunkMetaData meta, FSDataInputStream fin) throws IOException {
+    DictionaryPageHeader dictHeader = pageHeader.getDictionary_page_header();
+
+    int uncompressedPageSize = pageHeader.getUncompressed_page_size();
+    int compressedPageSize = pageHeader.getCompressed_page_size();
+
+    byte [] dictPageBytes = new byte[compressedPageSize];
+
+    fin.readFully(dictPageBytes);
 
+    BytesInput bin = BytesInput.from(dictPageBytes);
+
+    DictionaryPage compressedPage = new DictionaryPage(bin, uncompressedPageSize, dictHeader.getNum_values(), converter.getEncoding(dictHeader.getEncoding()));
+
+    BytesDecompressor decompressor = new CodecFactory(conf).getDecompressor(meta.getCodec());
+    DictionaryPageReader dictionaryPageReader = new ColumnChunkDictionaryPageReader(decompressor, compressedPage);
```

Comment:
A lot of this is carryover from https://github.com/apache/parquet-mr/pull/270 which fed into this one.  I believe the original implementation was something like this and the suggestion was to create something like the page reader implementation here.